UnitTests/MPRoktTests.m

```diff
@@ -0,0 +1,153 @@
+#import <XCTest/XCTest.h>
+#import <OCMock/OCMock.h>
+#import "MParticle.h"
+#import "MPKitContainer.h"
+#import "MPForwardQueueParameters.h"
+
+@interface MPRokt (Testing)
+@end
+
+@interface MPRoktTests : XCTestCase
+@property (nonatomic, strong) MPRokt *rokt;
+@end
+
+@implementation MPRoktTests
+
+- (void)setUp {
+    [super setUp];
+    self.rokt = [[MPRokt alloc] init];
+}
+
+- (void)tearDown {
+    self.rokt = nil;
+    [super tearDown];
+}
+
+- (void)testSelectPlacementsWithValidParameters {
+    id mockInstance = OCMClassMock([MParticle class]);
+    id mockContainer = OCMClassMock([MPKitContainer_PRIVATE class]);
+    [[[mockInstance stub] andReturn:mockContainer] kitContainer_PRIVATE];
+    [[[mockInstance stub] andReturn:mockInstance] sharedInstance];
+    
+    // Set up test parameters
+    NSString *viewName = @"testView";
+    NSDictionary *attributes = @{@"key": @"value"};
+    
+    // Set up expectations for kit container
+    OCMExpect([mockContainer forwardSDKCall:@selector(executeWithViewName:attributes:placements:onLoad:onUnLoad:onShouldShowLoadingIndicator:onShouldHideLoadingIndicator:onEmbeddedSizeChange:filteredUser:)
+                                              event:nil
+                                         parameters:[OCMArg checkWithBlock:^BOOL(MPForwardQueueParameters *params) {
+        XCTAssertEqualObjects(params[0], viewName);
+        XCTAssertNil(params[1]);
+        XCTAssertNil(params[2]);
+        XCTAssertNil(params[3]);
+        XCTAssertNil(params[4]);
+        XCTAssertNil(params[5]);
+        XCTAssertNil(params[6]);
+        XCTAssertNil(params[7]);
+        return true;
+    }]
+                                       messageType:MPMessageTypeEvent
+                                          userInfo:nil]);
+    
+    // Execute method
+    [self.rokt selectPlacements:viewName
+                     attributes:attributes];
+    
+    // Wait for async operation
+    XCTestExpectation *expectation = [self expectationWithDescription:@"Wait for async operation"];
+    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.1 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
+        [expectation fulfill];
+    });
+    [self waitForExpectationsWithTimeout:1.0 handler:nil];
+    
+    // Verify
+    OCMVerifyAll(mockContainer);
+}
+
+- (void)testExecuteWithValidParameters {
```

Comment:
it's no longer `Execute`, so should this also be `testSelectPlacements...`
```suggestion
- (void)testExecuteWithValidParameters {
```

and if so, should the 1st test here be something like `testSelectPlacementsWithOnlyRequiredParameters`

Reply:
So if you look at the MPKitProtocol.h I left the protocol method matching the Rokt SDK method because that method has all the possible parameters a client might want to pass in. That's also why I left the expanded version of the method in the Mparticle.Rokt as an additional option beyond the super simplified method we settled on.

I don't have any problem renaming the test but I want to make sure you understand wht I'm doing here/the choices I made.